### ISSUE TITLE
How to get the gas fee when transfer ton or jetton token 

### ISSUE THREAD

**Issue by 0xryanz** [xssnick/tonutils-go#281]
How to get the estimate gas fee when transfer ton or jetton token ?

**Comment by wythers** [comment]
Sorry, bro, theoretically, you can’t know the exact fee before the transaction is confirmed.

> Fees on TON are difficult to calculate in advance, as their amount depends on transaction run time, account status, message content and size, blockchain network settings, and a number of other variables that cannot be calculated until the transaction is sent.  

but there is a workaround method to estimate the fee from the history of similar transactions.
Some third-party APIs provide such methods, for example:  
[toncenter estimate fee API](https://toncenter.com/api/v2/#/send/estimate_fee_estimateFee_post)

**Comment by 0xryanz** [comment]
thanks

**Comment by aitimate** [comment]
> thanks

hi bro, do you solved it?

**Comment by wythers** [comment]
fwd fee plus storage fee = total 0.000494994 TON, seems to be no problem.

> > thanks
> 
> hi bro, do you solved it?

Em, fwd fee plus storage fee = total 0.000494994 TON, seems to be no problem.

**Comment by wythers** [comment]
> > thanks
> 
> hi bro, do you solved it?

It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet

**Comment by wythers** [comment]
> > > thanks
> > 
> > 
> > hi bro, do you solved it?
> 
> It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet

And the entire jetton line fee includes jetton transfer, jetton internal transfer, notify msg, and excess msg.
So, Total 0.000494994 TON, seems to be no problem.

**Comment by aitimate** [comment]
> no, see: https://tonviewer.com/transaction/cf1fcebfa405fc117affd8ab040cb3a9550d55d97f232ca49760cd82fca9047c?section=valueFlow

**Comment by aitimate** [comment]
> > > > thanks
> > > 
> > > 
> > > hi bro, do you solved it?
> > 
> > 
> > It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet
> 
> And the entire jetton line fee includes jetton transfer, jetton internal transfer, notify msg, and excess msg. So, Total 0.000494994 TON, seems to be no problem.

no, see: https://tonviewer.com/transaction/cf1fcebfa405fc117affd8ab040cb3a9550d55d97f232ca49760cd82fca9047c?section=valueFlow

The values differ by an order of magnitude of 10.

**Comment by wythers** [comment]
> > > > > thanks
> > > > 
> > > > 
> > > > hi bro, do you solved it?
> > > 
> > > 
> > > It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet
> > 
> > 
> > And the entire jetton line fee includes jetton transfer, jetton internal transfer, notify msg, and excess msg. So, Total 0.000494994 TON, seems to be no problem.
> 
> no, see: https://tonviewer.com/transaction/cf1fcebfa405fc117affd8ab040cb3a9550d55d97f232ca49760cd82fca9047c?section=valueFlow
> 
> The values differ by an order of magnitude of 10.

Bro, the estimate API is ONLY for normal transaction, The way you transfer jettons is not normal. The API takes the records of previous normal transactions to match similar transactions

**Comment by aitimate** [comment]
> > > > > > thanks
> > > > > 
> > > > > 
> > > > > hi bro, do you solved it?
> > > > 
> > > > 
> > > > It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet
> > > 
> > > 
> > > And the entire jetton line fee includes jetton transfer, jetton internal transfer, notify msg, and excess msg. So, Total 0.000494994 TON, seems to be no problem.
> > 
> > 
> > no, see: https://tonviewer.com/transaction/cf1fcebfa405fc117affd8ab040cb3a9550d55d97f232ca49760cd82fca9047c?section=valueFlow
> > The values differ by an order of magnitude of 10.
> 
> Bro, the estimate API is ONLY for normal transaction, The way you transfer jettons is not normal. The API takes the records of previous normal transactions to match similar transactions

ohw, I see tonweb.ts is surpported

**Comment by wythers** [comment]
> > > > > > > thanks
> > > > > > 
> > > > > > 
> > > > > > hi bro, do you solved it?
> > > > > 
> > > > > 
> > > > > It estimates the fee for the entire jetton line, not the fee for your wallet to your jetton wallet
> > > > 
> > > > 
> > > > And the entire jetton line fee includes jetton transfer, jetton internal transfer, notify msg, and excess msg. So, Total 0.000494994 TON, seems to be no problem.
> > > 
> > > 
> > > no, see: https://tonviewer.com/transaction/cf1fcebfa405fc117affd8ab040cb3a9550d55d97f232ca49760cd82fca9047c?section=valueFlow
> > > The values differ by an order of magnitude of 10.
> > 
> > 
> > Bro, the estimate API is ONLY for normal transaction, The way you transfer jettons is not normal. The API takes the records of previous normal transactions to match similar transactions
> 
> ohw, I see tonweb.ts is surpported

Nice,  Typescript Ton ecosystem is much better than golang. hahahahhhhh
